<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1253</width>
    <height>592</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PWRi</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab_set">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>120</y>
          <width>341</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Select save folder</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushSelectImg">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>80</y>
          <width>91</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushSaveImg">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>140</y>
          <width>91</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="line_load">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>80</y>
          <width>351</width>
          <height>27</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>60</y>
          <width>341</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Select first image</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="line_save">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>140</y>
          <width>351</width>
          <height>27</height>
         </rect>
        </property>
       </widget>
       <widget class="QSpinBox" name="firstNumber">
        <property name="geometry">
         <rect>
          <x>640</x>
          <y>80</y>
          <width>61</width>
          <height>31</height>
         </rect>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
       </widget>
       <widget class="QSpinBox" name="lastNumber">
        <property name="geometry">
         <rect>
          <x>790</x>
          <y>80</y>
          <width>61</width>
          <height>30</height>
         </rect>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>640</x>
          <y>60</y>
          <width>131</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>First image</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>790</x>
          <y>60</y>
          <width>141</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Last image</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="angleStart">
        <property name="geometry">
         <rect>
          <x>640</x>
          <y>140</y>
          <width>76</width>
          <height>30</height>
         </rect>
        </property>
        <property name="minimum">
         <double>-360.000000000000000</double>
        </property>
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="angleEnd">
        <property name="geometry">
         <rect>
          <x>790</x>
          <y>140</y>
          <width>76</width>
          <height>30</height>
         </rect>
        </property>
        <property name="minimum">
         <double>-360.000000000000000</double>
        </property>
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
       </widget>
       <widget class="QLineEdit" name="nImages">
        <property name="geometry">
         <rect>
          <x>950</x>
          <y>80</y>
          <width>61</width>
          <height>27</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>ArrowCursor</cursorShape>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLineEdit" name="angleStep">
        <property name="geometry">
         <rect>
          <x>950</x>
          <y>140</y>
          <width>121</width>
          <height>27</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>ArrowCursor</cursorShape>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>950</x>
          <y>60</y>
          <width>291</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Amount of images in the folder</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>640</x>
          <y>120</y>
          <width>131</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>First angle  (°)</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>790</x>
          <y>120</y>
          <width>161</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Last angle (°)</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>950</x>
          <y>120</y>
          <width>111</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Angle step  (°)</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>1201</width>
          <height>211</height>
         </rect>
        </property>
        <property name="title">
         <string>Images</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>230</y>
          <width>1201</width>
          <height>241</height>
         </rect>
        </property>
        <property name="title">
         <string>Parameters</string>
        </property>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>60</y>
           <width>68</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>Lens</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_19">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>110</y>
           <width>111</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Wavelength</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_20">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>160</y>
           <width>111</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>Prism angle</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_21">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>210</y>
           <width>111</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>Prism index</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_isCalibOK">
         <property name="geometry">
          <rect>
           <x>550</x>
           <y>170</y>
           <width>631</width>
           <height>31</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Calibration files does not exist. Please calibrate the system first.</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_lens">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>60</y>
           <width>113</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_wavelength">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>110</y>
           <width>113</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_angle">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>160</y>
           <width>113</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_index">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>210</y>
           <width>113</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>60</y>
           <width>21</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>110</y>
           <width>31</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>nm</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_14">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>160</y>
           <width>31</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>°</string>
         </property>
        </widget>
        <widget class="QPushButton" name="specificCalibration">
         <property name="geometry">
          <rect>
           <x>920</x>
           <y>100</y>
           <width>261</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Load a specific calibration file</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_loadSpecCalib">
         <property name="geometry">
          <rect>
           <x>550</x>
           <y>100</y>
           <width>351</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QCheckBox" name="checkBox_specificCalib">
         <property name="geometry">
          <rect>
           <x>550</x>
           <y>60</y>
           <width>261</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Use a specific calibration file</string>
         </property>
        </widget>
       </widget>
       <zorder>groupBox</zorder>
       <zorder>groupBox_2</zorder>
       <zorder>label_2</zorder>
       <zorder>pushSelectImg</zorder>
       <zorder>pushSaveImg</zorder>
       <zorder>line_load</zorder>
       <zorder>label</zorder>
       <zorder>line_save</zorder>
       <zorder>firstNumber</zorder>
       <zorder>lastNumber</zorder>
       <zorder>label_7</zorder>
       <zorder>label_8</zorder>
       <zorder>angleStart</zorder>
       <zorder>angleEnd</zorder>
       <zorder>nImages</zorder>
       <zorder>angleStep</zorder>
       <zorder>label_9</zorder>
       <zorder>label_10</zorder>
       <zorder>label_11</zorder>
       <zorder>label_12</zorder>
      </widget>
      <widget class="QWidget" name="tab_cal">
       <attribute name="title">
        <string>Calibration</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>1201</width>
          <height>461</height>
         </rect>
        </property>
        <property name="title">
         <string>Calibration</string>
        </property>
        <widget class="QPushButton" name="imageCalibButton">
         <property name="geometry">
          <rect>
           <x>740</x>
           <y>70</y>
           <width>91</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="line_imgCalib">
         <property name="geometry">
          <rect>
           <x>32</x>
           <y>70</y>
           <width>691</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>50</y>
           <width>431</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Select the first image of the calibration set</string>
         </property>
        </widget>
        <widget class="QPushButton" name="calibrateButton">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>290</y>
           <width>91</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Calibrate</string>
         </property>
        </widget>
        <widget class="QLabel" name="stateLabel">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>290</y>
           <width>421</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_defaultParameters">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>130</y>
           <width>291</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>I want to use the default parameters</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QRadioButton" name="radioButton_specificName">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>170</y>
           <width>471</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>I want to use specific name and directory for my calibration</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QLineEdit" name="line_saveSpecific">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>220</y>
           <width>351</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QProgressBar" name="calibBar">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>292</y>
           <width>671</width>
           <height>21</height>
          </rect>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_preproc">
       <attribute name="title">
        <string>Pre-processing</string>
       </attribute>
       <widget class="QProgressBar" name="spectreBar">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>660</x>
          <y>370</y>
          <width>541</width>
          <height>20</height>
         </rect>
        </property>
        <property name="mouseTracking">
         <bool>false</bool>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>621</width>
          <height>171</height>
         </rect>
        </property>
        <property name="title">
         <string>Pre-processing tools</string>
        </property>
        <widget class="QPushButton" name="preproc">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>50</y>
           <width>181</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Undistortion</string>
         </property>
        </widget>
        <widget class="QPushButton" name="getExtrema">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>110</y>
           <width>181</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Get Extrema</string>
         </property>
        </widget>
        <widget class="QProgressBar" name="preprocBar">
         <property name="geometry">
          <rect>
           <x>220</x>
           <y>50</y>
           <width>381</width>
           <height>31</height>
          </rect>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>190</y>
          <width>621</width>
          <height>271</height>
         </rect>
        </property>
        <property name="title">
         <string>Resonance images</string>
        </property>
        <widget class="QLabel" name="imageP">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>40</y>
           <width>271</width>
           <height>201</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="imageS">
         <property name="geometry">
          <rect>
           <x>330</x>
           <y>40</y>
           <width>271</width>
           <height>201</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="P_label">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>240</y>
           <width>281</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="S_label">
         <property name="geometry">
          <rect>
           <x>330</x>
           <y>240</y>
           <width>271</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_6">
        <property name="geometry">
         <rect>
          <x>649</x>
          <y>9</y>
          <width>561</width>
          <height>451</height>
         </rect>
        </property>
        <property name="title">
         <string>Spectrum</string>
        </property>
        <widget class="QCustomPlot" name="customPlot" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>541</width>
           <height>301</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="afficheSpectre">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>390</y>
           <width>131</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Draw Spectrum</string>
         </property>
        </widget>
        <widget class="QPushButton" name="SaveSpectrum">
         <property name="geometry">
          <rect>
           <x>340</x>
           <y>390</y>
           <width>131</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Save Spectrum</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btn_testSpectrePixel">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>420</y>
           <width>131</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Test spectrum</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btn_SaveRawData">
         <property name="geometry">
          <rect>
           <x>340</x>
           <y>420</y>
           <width>131</width>
           <height>30</height>
          </rect>
         </property>
         <property name="text">
          <string>Save Raw Data</string>
         </property>
        </widget>
       </widget>
       <zorder>groupBox_6</zorder>
       <zorder>groupBox_5</zorder>
       <zorder>groupBox_4</zorder>
       <zorder>spectreBar</zorder>
      </widget>
      <widget class="QWidget" name="tab_proc">
       <attribute name="title">
        <string>Processing</string>
       </attribute>
       <widget class="QLabel" name="resultImage">
        <property name="geometry">
         <rect>
          <x>230</x>
          <y>100</y>
          <width>401</width>
          <height>321</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="x1">
        <property name="geometry">
         <rect>
          <x>720</x>
          <y>60</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLabel" name="y1">
        <property name="geometry">
         <rect>
          <x>720</x>
          <y>100</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLabel" name="x2">
        <property name="geometry">
         <rect>
          <x>830</x>
          <y>60</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLabel" name="y2">
        <property name="geometry">
         <rect>
          <x>830</x>
          <y>100</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_18">
        <property name="geometry">
         <rect>
          <x>720</x>
          <y>20</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>From</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_22">
        <property name="geometry">
         <rect>
          <x>830</x>
          <y>20</y>
          <width>68</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>To</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_save">
       <attribute name="title">
        <string>Save</string>
       </attribute>
       <widget class="QPushButton" name="createVideo">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>30</y>
          <width>181</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Create Video</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1253</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
